### ISSUE TITLE
WC 6.9: array_filter called on woocommerce_enqueue_styles result produces fatal error with PHP 8 and some themes

### ISSUE THREAD

**Issue by ecelticuser** [woothemes/woocommerce#34667]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I've had woocommerce wordpress plugin for years and regularly updated - on my most recent update to 6.9, my website displayed the following warning below; I had to rollback to version 6.8.2 - the latest version 6.9 has a bug and needs to be resolved asap
 
Warning: array_filter() expects parameter 1 to be array, bool given in ..../woocommerce/includes/class-wc-frontend-scripts.php on line 65

### Expected behavior

there shoudl be no warning displayed on a website

### Actual behavior

see aboveI've had woocommerce wordpress plugin for years and regularly updated - on my most recent update to 6.9, my website displayed the following warning below; I had to rollback to version 6.8.2 - the latest version 6.9 has a bug and needs to be resolved asap
 
Warning: array_filter() expects parameter 1 to be array, bool given in ..../woocommerce/includes/class-wc-frontend-scripts.php on line 65

### Steps to reproduce

1) update to 6.9 version
2) open a new window and check a website
3) top line teh warning is displayed

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://ashburybloom.ca/web
Site address (URL): https://ashburybloom.ca
WC Version: 6.8.2
REST API Version: ✔ 6.8.2
WC Blocks Version: ✔ 8.0.0
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.30
PHP Post Max Size: 30 MB
PHP Time Limit: 200
PHP Max Input Vars: 1000
cURL Version: 7.61.1
OpenSSL/1.1.1k

SUHOSIN Installed: –
MySQL Version: 8.0.30
Max Upload Size: 30 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.9.0
WC Database Prefix: wp_
Total Database Size: 17.27MB
Database Data Size: 14.01MB
Database Index Size: 3.26MB
wp_woocommerce_sessions: Data: 5.30MB + Index: 0.14MB + Engine MyISAM
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_order_items: Data: 0.10MB + Index: 0.06MB + Engine MyISAM
wp_woocommerce_order_itemmeta: Data: 0.81MB + Index: 0.50MB + Engine MyISAM
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_actionscheduler_actions: Data: 0.12MB + Index: 0.07MB + Engine MyISAM
wp_actionscheduler_claims: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_actionscheduler_groups: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_actionscheduler_logs: Data: 0.08MB + Index: 0.06MB + Engine MyISAM
wp_commentmeta: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_comments: Data: 0.26MB + Index: 0.10MB + Engine MyISAM
wp_e_events: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_frmt_form_entry: Data: 0.04MB + Index: 0.05MB + Engine MyISAM
wp_frmt_form_entry_meta: Data: 1.27MB + Index: 0.37MB + Engine MyISAM
wp_frmt_form_views: Data: 0.07MB + Index: 0.12MB + Engine MyISAM
wp_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_mailchimp_carts: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_mailchimp_jobs: Data: 0.01MB + Index: 0.00MB + Engine MyISAM
wp_ms_snippets: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_nextend2_image_storage: Data: 0.02MB + Index: 0.01MB + Engine MyISAM
wp_nextend2_section_storage: Data: 0.03MB + Index: 0.01MB + Engine MyISAM
wp_nextend2_smartslider3_generators: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_nextend2_smartslider3_sliders: Data: 0.05MB + Index: 0.00MB + Engine MyISAM
wp_nextend2_smartslider3_sliders_xref: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_nextend2_smartslider3_slides: Data: 0.06MB + Index: 0.01MB + Engine MyISAM
wp_options: Data: 1.78MB + Index: 0.23MB + Engine MyISAM
wp_pimwick_gift_card: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_pimwick_gift_card_activity: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_postmeta: Data: 2.74MB + Index: 0.88MB + Engine MyISAM
wp_posts: Data: 0.48MB + Index: 0.13MB + Engine MyISAM
wp_sfba_subscribers_lists: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_sfba_subscription_lists: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_snippets: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_term_relationships: Data: 0.01MB + Index: 0.03MB + Engine MyISAM
wp_term_taxonomy: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_termmeta: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_terms: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_tm_taskmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_tm_tasks: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_usermeta: Data: 0.21MB + Index: 0.12MB + Engine MyISAM
wp_users: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.01MB + Engine MyISAM
wp_wc_admin_notes: Data: 0.04MB + Index: 0.00MB + Engine MyISAM
wp_wc_category_lookup: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.02MB + Engine MyISAM
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_order_coupon_lookup: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_order_product_lookup: Data: 0.11MB + Index: 0.10MB + Engine MyISAM
wp_wc_order_stats: Data: 0.03MB + Index: 0.02MB + Engine MyISAM
wp_wc_order_tax_lookup: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_tax_rate_classes: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpforms_tasks_meta: Data: 0.20MB + Index: 0.00MB + Engine MyISAM
wp_wtbp_columns: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wtbp_modules: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wtbp_modules_type: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wtbp_tables: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wtbp_usage_stat: Data: 0.00MB + Index: 0.00MB + Engine MyISAM

### Post Type Counts ###

attachment: 617
custom_css: 2
elementor_library: 1
envira: 1
foogallery: 1
forminator_forms: 3
mc4wp-form: 1
nav_menu_item: 23
page: 23
pluginops_forms: 2
post: 1
product: 83
product_variation: 23
revision: 32
shop_coupon: 17
shop_order: 381
shop_order_refund: 5
wpforms: 4

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (18) ###

Canada Post Shipping For WooCommerce: by Small Fish Analytics Inc. – 2.9.8
Classic Editor: by WordPress Contributors – 1.6.2
Classic Widgets: by WordPress Contributors – 0.3
Custom Order Numbers for WooCommerce: by Tyche Softwares – 1.3.5
Elementor: by Elementor.com – 3.7.4
Forminator: by WPMU DEV – 1.17.2
Mailchimp for WooCommerce: by Mailchimp – 2.7.5
MC4WP: Mailchimp for WordPress: by ibericode – 4.8.8
Ocean Extra: by OceanWP – 2.0.4
Ocean Product Sharing: by OceanWP – 2.0.4
Ocean Social Sharing: by OceanWP – 2.0.2
Simple Custom CSS: by John Regan – 4.0.5
Smart Slider 3: by Nextend – 3.5.1.9
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 1.22.21

WooCommerce Stripe Gateway: by WooCommerce – 6.7.0
WooCommerce: by Automattic – 6.8.2 (update to version 6.9.0 is available)
WP-Optimize - Clean, Compress, Cache: by David Anderson
Ruhani Rabin
Team Updraft – 3.2.7

WP Rollback: by GiveWP.com – 1.7.2

### Inactive Plugins (1) ###

Password Protected: by WPExperts – 2.5.3

### Dropin Plugins (1) ###

maintenance.php: maintenance.php

### Settings ###

API Enabled: –
Force SSL: –
Currency: CAD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
gift-card (gift-card)
grouped (grouped)
pw gift card (pw-gift-card)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: #564 - /terms-of-service/

### Theme ###

Name: OceanWP
Version: 3.3.4
Author URL: https://oceanwp.org/about-oceanwp/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: oceanwp/woocommerce/cart/mini-cart.php
oceanwp/woocommerce/content-single-product.php
oceanwp/woocommerce/loop/loop-start.php
oceanwp/woocommerce/single-product/title.php

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page
wc-pay-subscriptions-page

Disabled Features: minified-js
settings

Daily Cron: ✔ Next scheduled: 2022-09-14 08:01:15 -04:00
Options: ✔
Notes: 52
Onboarding: completed

### Action Scheduler ###

Complete: 67
Oldest: 2022-08-14 14:38:33 -0400
Newest: 2022-09-13 22:51:08 -0400

Pending: 1
Oldest: 2022-09-14 23:02:42 -0400
Newest: 2022-09-14 23:02:42 -0400

### Status report information ###

Generated at: 2022-09-13 23:34:23 -04:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by lazehang** [comment]
any updates? facing the same issue.

**Comment by BrentOzar** [comment]
Same here. In log:

2022-09-14T04:26:15+00:00 CRITICAL Uncaught TypeError: array_filter(): Argument #1 ($array) must be of type array, bool given in /nas/content/live/sitename/wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php:65

**Comment by HILAYTRIVEDI** [comment]
I found this piece of code here, that raises this error. 
`return array_filter(
			apply_filters(
				'woocommerce_enqueue_styles',
				array(
					'woocommerce-layout'      => array(
						'src'     => self::get_asset_url( 'assets/css/woocommerce-layout.css' ),
						'deps'    => '',
						'version' => $version,
						'media'   => 'all',
						'has_rtl' => true,
					),
					'woocommerce-smallscreen' => array(
						'src'     => self::get_asset_url( 'assets/css/woocommerce-smallscreen.css' ),
						'deps'    => 'woocommerce-layout',
						'version' => $version,
						'media'   => 'only screen and (max-width: ' . apply_filters( 'woocommerce_style_smallscreen_breakpoint', '768px' ) . ')',
						'has_rtl' => true,
					),
					'woocommerce-general'     => array(
						'src'     => self::get_asset_url( 'assets/css/woocommerce.css' ),
						'deps'    => '',
						'version' => $version,
						'media'   => 'all',
						'has_rtl' => true,
					),
					'woocommerce-blocktheme'  => wc_current_theme_is_fse_theme() ? array(
						'src'     => self::get_asset_url( 'assets/css/woocommerce-blocktheme.css' ),
						'deps'    => '',
						'version' => $version,
						'media'   => 'all',
						'has_rtl' => true,
					) : false,
				)
			)
		);`

But keeping that aside, I can't see warnings on my site, It is a fresh setup of WooCommerce 6.9.0. I am also attaching the screen recording for the same.

https://www.awesomescreenshot.com/video/11074213?key=d304c3af92d8e9af64f148da2c3fe761

Am I missing something here?

**Comment by pau1phi11ips** [comment]
I've downgraded PHP from 8.0 to 7.4 on our site and it's kinda working again. 

PHP is just throwing Warnings instead of Critical errors now so at least it's functional again.

**Comment by pau1phi11ips** [comment]
Here's the stack trace for the error

```
AH01071: Got error 'PHP message: PHP Fatal error: Uncaught TypeError: array_filter(): Argument #1 ($array) must be of type array, bool given in /wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php:65
Stack trace:
#0 /wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php(65): array_filter()
#1 /wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php(430): WC_Frontend_Scripts::get_styles()
#2 /wp-includes/class-wp-hook.php(307): WC_Frontend_Scripts::load_scripts()
#3 /wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters()
#4 /wp-includes/plugin.php(476): WP_Hook->do_action()
#5 /wp-includes/script-loader.php(2156): do_action()
#6 /wp-includes/class-wp-hook.php(307): wp_enqueue_scripts()
#7 ...', referer: https://site.com/wp-admin/update-core.php?action=do-theme-upgrade
```

**Comment by kuba14** [comment]
Facing the same issue.

**Comment by david-strejc** [comment]
Fastest quick fix:


```
57         public static function get_styles() {
 58                 $version = Constants::get_constant( 'WC_VERSION' );
 59 
```

after line 57 in /wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php

add:

`return true;`

so it looks like:
```

 57         public static function get_styles() {
 58                 return true;
 59                 $version = Constants::get_constant( 'WC_VERSION' );
 60 
```

**Comment by heinzoellrich** [comment]
same issue on several woocommerce sites
had to rollback
php8 and 8.1 and 7.4

**Comment by sukafia** [comment]
Reported in 36909764-hc and 36918673-hc

**Comment by ibndawood** [comment]
Please add this to your child theme's functions.php to temporarily resolve this issue:

```
add_filter( 'woocommerce_enqueue_styles', 'wc_child_theme_fix_33467', PHP_INT_MAX );

function wc_child_theme_fix_33467( $styles ) {
    return ( is_array( $styles ) ) ? $styles : [];
}
```

**Comment by lazehang** [comment]
Till a patch comes out, you can downgrade woocommerce back to v6.8.2.

Steps to downgrade would be:
1. Backup the whole site.
2. Download woocommerce plugin zip (6.8.2) from [here](https://developer.woocommerce.com/releases/)
3. Go to Plugins > Add New and upload the zip file that was downloaded in step 2 and Install it.

**Comment by kaushikasomaiya** [comment]
5555958-zen, 5556022-zen

**Comment by peterfabian** [comment]
PR this was introduced in: https://github.com/woocommerce/woocommerce/pull/33518

**Comment by kaushikasomaiya** [comment]
5556064-zen (OceanWP)

**Comment by neilmccreadie** [comment]
5556160-zen

**Comment by thuautp** [comment]
5556344-zen

**Comment by peterfabian** [comment]
OceanWP published an updated version (3.3.5) about 3 hours ago, but other themes might still be affected.

**Comment by JosefHrib** [comment]
I have same fatal error. Last WP, WC, Theme Motors, PHP 8.0.23.

I used fast fix.
after line 57 in /wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php
add:
return true;
so it looks like:
 57         public static function get_styles() {
 58                 return true;
 59                 $version = Constants::get_constant( 'WC_VERSION' );
 60

Frontend works, but on frontend is still "Warning".
Warning: foreach() argument must be of type array|object, bool given in /www/doc/www.karavantravel.cz/www/wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php on line 433

**Comment by stuartduff** [comment]
5556745-zen

**Comment by WinISOpl** [comment]
> I have same fatal error. Last WP, WC, Theme Motors, PHP 8.0.23.
> 
> I used fast fix. after line 57 in /wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php add: return true; so it looks like: 57 public static function get_styles() { 58 return true; 59 $version = Constants::get_constant( 'WC_VERSION' ); 60
> 
> Frontend works, but on frontend is still "Warning". Warning: foreach() argument must be of type array|object, bool given in /www/doc/www.karavantravel.cz/www/wp-content/plugins/woocommerce/includes/class-wc-frontend-scripts.php on line 433

I have exactly the same

**Comment by stuartduff** [comment]
5556727-zen

**Comment by amElnagdy** [comment]
5557089-zen

**Comment by pau1phi11ips** [comment]
@JosefHrib This is a better fix [#issuecomment-1246458316](https://github.com/woocommerce/woocommerce/issues/34667#issuecomment-1246458316)

**Comment by thelogicwizards** [comment]
[5557513-zen](https://woothemes.zendesk.com/agent/tickets/5557513)
